oshmem/proc/proc.h

```diff
@@ -35,30 +37,11 @@ struct oshmem_group_t;
 
 #define OSHMEM_PE_INVALID   (-1)
 
-/**
- * Remote Open SHMEM process structure
- *
- * Remote Open SHMEM process structure.  Each process contains exactly
- * one oshmem_proc_t structure for each remote process it knows about.
- */
-struct oshmem_proc_t {
-    opal_proc_t                    super;
-    /* endpoint data */
-    void *proc_endpoints[OMPI_PROC_ENDPOINT_TAG_MAX];
-    /*
-     * All transport channels are globally ordered.
-     * pe(s) can talk to each other via subset of transports
-     * these holds indexes of each transport into global array
-     *  proc -> id, where id can be btl id in yoda or mxm ptl id
-     *  in ikrit
-     *  spml is supposed to fill this during add_procs()
-     **/
-    int                             num_transports;
-    char                           *transport_ids;
-};
+#define OSHMEM_PROC_NUM_TRANSPORTS(proc) \
+    (*(int *)(proc)->padding)
 
-typedef struct oshmem_proc_t oshmem_proc_t;
-OBJ_CLASS_DECLARATION(oshmem_proc_t);
+#define OSHMEM_PROC_TRANSPORT_IDS(proc) \
+    (*(char **)(proc->padding + 8))
```

Comment:
Can we do something not based on a fixed byte offset, such as casting to a struct that fits into the padding?  This would also give a nice way to check that the padding is `>=` than the content you want to put into the padding (for the accidental tautological sizeof comparison).